projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -85,13 +96,43 @@ export class CartesianLegend {
   }
 
   private drawLegendEntries(container: ContainerElement): void {
-    select(container)
-      .append('div')
-      .classed('legend-entries', true)
-      .selectAll('.legend-entry')
-      .data(this.series.filter(series => !series.hide))
-      .enter()
-      .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    const containerSelection = select(container);
+    if (!this.isGrouped) {
+      containerSelection
+        .append('div')
+        .classed('legend-entries', true)
+        .selectAll('.legend-entry')
+        .data(this.series.filter(series => !series.hide))
+        .enter()
+        .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    } else {
+      containerSelection
+        .selectAll('.legend-entries')
```

Comment:
We should try to use d3 update pattern meaning we should add merge and exit calls as well. But we can do it later

Reply:
The way we're using this component/class, it doesn't matter actually so whenever we change the series, cartesian chart will create a new instance of the legend, That should be fixed first.

Reply:
that's true.